src/stackedviews.jl

```diff
@@ -159,3 +159,16 @@ firstinds() = error("not all arrays can be zeroarray")
 @inline take_zeros(T, inds, ::ZeroArrayPromise, Bs...) = (ZeroArrayPromise{T}(inds), take_zeros(T, inds, Bs...)...)
 @inline take_zeros(T, inds, A::AbstractArray, Bs...) = (A, take_zeros(T, inds, Bs...)...)
 take_zeros(T, inds) = ()
+
+# Overrides for paddedviews
+function PaddedViews.paddedviews(fillvalue, As::Union{AbstractArray,ZeroArrayPromise}...)
+    inds = PaddedViews.outerinds(As...)
+    map(A->PaddedView(fillvalue, A, inds), As)
+end
+
+(::Type{PaddedViews.PaddedView})(fillvalue, zap::ZeroArrayPromise, indices) = zap
+
+@inline PaddedViews.outerinds(A::ZeroArrayPromise, Bs...) = PaddedViews.outerinds(Bs...)
+@inline PaddedViews.outerinds() = error("must supply at least one array with concrete indices")
```

Comment:
I wondered the same thing myself. The reason it's here is because I intended it to be impossible to reach this condition purely within PaddedViews, which restricts itself to `AbstractArray` inputs; it becomes possible here because of the dropping of `ZeroArrayPromise`, and the possibility that the user supplied only `ZeroArrayPromise`s.

However, I just noticed this:
```julia
julia> using PaddedViews

julia> pv = paddedviews(2)
ERROR: MethodError: no method matching outerinds()
Closest candidates are:
  outerinds(::AbstractArray{T,N}, ::Any...) at /home/tim/.julia/v0.5/PaddedViews/src/PaddedViews.jl:105
 in paddedviews(::Int64) at /home/tim/.julia/v0.5/PaddedViews/src/PaddedViews.jl:101
```
so clearly I messed up. PaddedViews should allow "empty" inputs and happily return `nothing` in this case.

I'm still inclined to leave this here because the error message only really makes sense in the context of someone supplying only `ZeroArrayPromise` inputs.